test/modules/aws-ecr-build-integration.ts

```diff
@@ -0,0 +1,128 @@
+import * as iasql from '../../src/services/iasql';
+import {
+  getPrefix,
+  runQuery,
+  runApply,
+  finish,
+  execComposeUp,
+  execComposeDown,
+  runSync,
+  runInstall,
+  runUninstall,
+  getKeyCertPair,
+} from '../helpers';
+import { AWS } from '../../src/services/aws_macros';
+
+const prefix = getPrefix();
+const dbAlias = 'ecrbuildertest';
+const repositoryName = prefix + dbAlias;
+
+
+const apply = runApply.bind(null, dbAlias);
+const sync = runSync.bind(null, dbAlias);
+const query = runQuery.bind(null, dbAlias);
+const install = runInstall.bind(null, dbAlias);
+const uninstall = runUninstall.bind(null, dbAlias);
+const modules = ['aws_ecr', 'aws_iam'];
+
+
+jest.setTimeout(360000);
+beforeAll(async () => await execComposeUp());
+afterAll(async () => await execComposeDown());
+
+describe('AwsEcrBuild Integration Testing', () => {
+  it('creates a new test db', (done) => void iasql.connect(
+    dbAlias,
+    'not-needed', 'not-needed').then(...finish(done)));
+
+  it('installs the aws_account module', install(['aws_account']));
+
+  it('inserts aws credentials', query(`
+      INSERT INTO aws_credentials (access_key_id, secret_access_key)
+      VALUES ('${process.env.AWS_ACCESS_KEY_ID}', '${process.env.AWS_SECRET_ACCESS_KEY}')
+  `, undefined, false));
+
+  it('syncs the regions', sync());
+
+  it('sets the default region', query(`
+      UPDATE aws_regions
+      SET is_default = TRUE
+      WHERE region = '${process.env.AWS_REGION}';
+  `));
+
+  it('installs the ecr module', install(modules));
+
+  it('creates a new ecr repository', query(`
+      INSERT INTO repository (repository_name, scan_on_push, image_tag_mutability)
+      VALUES ('${repositoryName}', false, 'MUTABLE');
+  `));
+
+  it('applies the creation of ecr repository', apply());
+
+  it('builds hello world image and pushes to the new ecr repo', query(`
+      SELECT ecr_build(
+                     'https://github.com/mtp1376/docker-helloworld',
```

Comment:
can we put this under the iasql org?

Reply:
Moved to https://github.com/iasql/docker-helloworld.